test/Microsoft.TemplateEngine.IDE.IntegrationTests/Utils/IFileChangeEqualityComparer.cs

```diff
@@ -0,0 +1,46 @@
+using Microsoft.TemplateEngine.Abstractions;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+
+namespace Microsoft.TemplateEngine.IDE.IntegrationTests.Utils
+{
+    internal class IFileChangeComparer : IEqualityComparer<IFileChange>, IComparer<IFileChange>
+    {
+        public int Compare(IFileChange x, IFileChange y)
+        {
+            if (Equals(x,y))
+            {
+                return 0;
+            }
+
+            if (x.ChangeKind > y.ChangeKind)
+            {
+                return 1;
+            }
+            if (x.ChangeKind < y.ChangeKind)
+            {
+                return -1;
+            }
+
+            if (!string.Equals(x.TargetRelativePath, y.TargetRelativePath, StringComparison.OrdinalIgnoreCase))
```

Comment:
Instead of checking with `Equals` first, we could directly call `Compare` and return if it is nonzero.